src/shared/install.c

```diff
@@ -1492,7 +1492,7 @@ static int install_info_symlink_alias(
                 rp = skip_root(paths, i->path);
 
                 q = create_symlink(rp ?: i->path, alias_path, force, changes, n_changes);
-                if (r == 0)
+                if (q < 0)
                         r = q;
```

Comment:
the old code was right here. The logic is to return the first error that happens, but continue operation. Thus, only when r is 0 we should override it.

(note that OOM is considered fatal here, only the symlink creation is something where we proceed after the first failure, but still return the first failure)